src/sqlfluff/core/parser/parsers.py

```diff
@@ -100,6 +100,43 @@ def match(
         return MatchResult.from_unmatched(segments)
 
 
+class MultiStringParser(StringParser):
+    """An object which matches and returns raw segments on a collection of strings."""
+
+    def __init__(
+        self,
+        templates: Collection[str],
+        raw_class: Type[RawSegment],
+        name: Optional[str] = None,
+        type: Optional[str] = None,
+        optional: bool = False,
+        **segment_kwargs,
+    ):
+        self.templates = templates
+        self.raw_class = raw_class
+        self.name = name
+        self.type = type
+        self.optional = optional
+        self.segment_kwargs = segment_kwargs or {}
+
+    def simple(self, parse_context: "ParseContext") -> Optional[List[str]]:
+        """Return simple options for this matcher.
+
+        Because string matchers are not case sensitive we can
+        just return the template here.
+        """
+        return [template.upper() for template in self.templates]
+
+    def _is_first_match(self, segment: BaseSegment):
+        """Does the segment provided match according to the current rules."""
+        # Is the target a match and IS IT CODE.
+        # The latter stops us accidentally matching comments.
+        upper_templates = {template.upper() for template in self.templates}
+        if segment.raw.upper() in upper_templates and segment.is_code:
+            return True
```

Comment:
Can remove need for upper_templates if uppering at init.

Also would it be slightly quicker to do the `is_code` check first? Especially for larger lists?

```py
if segment.is_code and segment.raw.upper() in upper_templates:
```